In [1]:
library(tidyverse)
library(testthat)
library(digest)
library(repr)
library(tidymodels)
options(repr.matrix.max.rows = 6)

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──

✔ ggplot2 3.3.6     ✔ purrr   0.3.4
✔ tibble  3.1.7     ✔ dplyr   1.0.9
✔ tidyr   1.2.0     ✔ stringr 1.4.0
✔ readr   2.1.2     ✔ forcats 0.5.1

── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()


Attaching package: ‘testthat’


The following object is masked from ‘package:dplyr’:

    matches


The following object is masked from ‘package:purrr’:

    is_null


The following objects are masked from ‘package:readr’:

    edition_get, local_edition


The following object is masked from ‘package:tidyr’:

    matches


── Attaching packages ────────────────────────────────────── tidymodels 1.0.0 ──

✔ broom        1.0.0     ✔ rsample      1.0.0
✔ dials        1.0.0     ✔ tune         1.0.0
✔ infer        1.0.2     ✔ workflows    1.0.0
✔ modeldata    1.0.0     ✔ workflowsets 1.0.0
✔ parsnip      1.0.0  

In [12]:
pulsar_data <- read_csv("https://raw.githubusercontent.com/KoiYouu/dcsi100_group_project/main/HTRU2/HTRU_2.csv", col_names = FALSE)
colnames(pulsar_data) <- c("Mean_of _the_integrated_profile", 
                           "Standard_deviation_of_the_integrated_profile",
                           "Excess_kurtosis_of_the_integrated_profile",
                           "Skewness_of_the_integrated_profile",
                           "Mean_of_the_DM-SNR_curve",
                           "Standard_deviation_of_the_DM-SNR_curve",
                           "Excess_kurtosis_of_the_DM-SNR_curve",
                           "Skewness_of_the_DM-SNR_curve",
                           "Class")
pulsar_data

Rows: 17898 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
dbl (9): X1, X2, X3, X4, X5, X6, X7, X8, X9

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Mean_of _the_integrated_profile
1     140.5625                       
2     102.5078                       
3     103.0156                       
⋮     ⋮                              
17896 119.3359                       
17897 114.5078                       
17898  57.0625                       
      Standard_deviation_of_the_integrated_profile
1     55.68378                                    
2     58.88243                                    
3     39.34165                                    
⋮     ⋮                                           
17896 59.93594                                    
17897 53.90240                                    
17898 85.79734                                    
      Excess_kurtosis_of_the_integrated_profile
1     -0.2345714                               
2      0.4653182                               
3      0.3233284                               
⋮     ⋮                                        
17896 0.1593631                                
17897 0.2011614                                
17898 1.4063910                                
      Skewness_of_the_integrated_profile Mean_of_the_DM-SNR_curve
1     -0.6996484                         3.199833                
2     -0.5150879                         1.677258                
3      1.0511644                         3.121237                
⋮     ⋮                                  ⋮                       
17896 -0.74302540                         21.430602              
17897 -0.02478884                          1.946488              
17898  0.08951971                        188.306020              
      Standard_deviation_of_the_DM-SNR_curve
1     19.11043                              
2     14.86015                              
3     21.74467                              
⋮     ⋮                                     
17896 58.87200                              
17897 13.38173                              
17898 64.71256                              
      Excess_kurtosis_of_the_DM-SNR_curve Skewness_of_the_DM-SNR_curve Class
1      7.975532                            74.24222                    0    
2     10.576487                           127.39358                    0    
3      7.735822                            63.17191                    0    
⋮     ⋮                                   ⋮                            ⋮    
17896  2.499517                             4.595173                   0    
17897 10.007967                           134.238910                   0    
17898 -1.597527                             1.429475                   0

Adding column headers to tidy up data


In [14]:
colnames(pulsar_data) <- c('mean_profile', 'std_profile', 'kurtosis_profile', 'skewness_profile', 'mean_dmsnr',
               'std_dmsnr', 'kurtosis_dmsnr', 'skewness_dmsnr', 'class')
pulsar_data

mean_profile std_profile kurtosis_profile skewness_profile mean_dmsnr
1     140.5625     55.68378    -0.2345714       -0.6996484       3.199833  
2     102.5078     58.88243     0.4653182       -0.5150879       1.677258  
3     103.0156     39.34165     0.3233284        1.0511644       3.121237  
⋮     ⋮            ⋮           ⋮                ⋮                ⋮         
17896 119.3359     59.93594    0.1593631        -0.74302540       21.430602
17897 114.5078     53.90240    0.2011614        -0.02478884        1.946488
17898  57.0625     85.79734    1.4063910         0.08951971      188.306020
      std_dmsnr kurtosis_dmsnr skewness_dmsnr class
1     19.11043   7.975532       74.24222      0    
2     14.86015  10.576487      127.39358      0    
3     21.74467   7.735822       63.17191      0    
⋮     ⋮         ⋮              ⋮              ⋮    
17896 58.87200   2.499517        4.595173     0    
17897 13.38173  10.007967      134.238910     0    
17898 64.71256  -1.597527        1.429475     0

Renaming column headers to be easier to use when coding

In [15]:
pulsar_split <- initial_split(pulsar_data, prop = 3/4, strata = class)  
pulsar_train <- training(pulsar_split)   
pulsar_test <- testing(pulsar_split)

Creating the training / testing split

In [17]:
pulsar_info <- pulsar_train %>%
    group_by(class) %>%
    summarize(n = n())
pulsar_info

class n    
1 0     12172
2 1      1251

We can see that there are 12172 observations that aren't pulsar stars, and 1251 that are in our dataset.
